images/airflow/2.9.2/python/mwaa/config/airflow.py

```diff
@@ -272,6 +272,14 @@ def _get_essential_airflow_api_config() -> Dict[str, str]:
     if os.environ.get("MWAA__CORE__AUTH_TYPE", "").lower() == "none":
         api_config["AIRFLOW__API__AUTH_BACKENDS"] = "airflow.api.auth.backend.default"
 
+    if os.environ.get("MWAA__API__ALLOWED_HEADERS"):
+        api_config["AIRFLOW__API__ACCESS_CONTROL_ALLOW_HEADERS"] = os.environ.get("MWAA__API__ALLOWED_HEADERS") 
+    if os.environ.get("MWAA__API__ALLOWED_METHODS"):
+        api_config["AIRFLOW__API__ACCESS_CONTROL_ALLOW_METHODS"] = os.environ.get("MWAA__API__ALLOWED_METHODS")
+    if os.environ.get("MWAA__API__ALLOWED_ORIGINS"):
+        api_config["AIRFLOW__API__ACCESS_CONTROL_ALLOW_ORIGINS"] = os.environ.get("MWAA__API__ALLOWED_ORIGINS")
```

Comment:
We don't need to create a corresponding MWAA configuration for every Airflow configuration. You can simply make use of the [`MWAA__CORE__CUSTOM_AIRFLOW_CONFIGS`](https://github.com/aws/amazon-mwaa-docker-images/blob/main/images/airflow/2.9.2/python/mwaa/config/airflow.py#L96).

Reply:
To elaborate, we have a set of Airflow configuration that are either essential (e.g. Celery config) or we are opinionated about, and so we define them explicitly. Everything else, however, is up to the user to define using the CUSTOM_AIRFLOW_CONFIGS environment variable.